internal/oci/annotations.go

```diff
@@ -216,4 +216,17 @@ const (
 
 	// AnnotationSecurityPolicy is used to specify a security policy for opengcs to enforce
 	AnnotationSecurityPolicy = "io.microsoft.virtualmachine.lcow.securitypolicy"
+	// AnnotationProcessDumpLocation specifies a path inside of containers to save process dumps to. As
+	// the scratch space for a container is generally cleaned up after exit, this is best set to a volume mount of
+	// some kind (vhd, bind mount, fileshare mount etc.)
+	AnnotationProcessDumpLocation = "io.microsoft.processdumplocation"
+
+	// AnnotationWCOWProcessDumpType specifies the type of dump to create when generating a local user mode
+	// process dump for Windows containers. The supported options are "mini", and "full".
+	// See DumpType: https://docs.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps
+	AnnotationWCOWProcessDumpType = "io.microsoft.wcow.processdumptype"
```

Comment:
Why not have these as shim options?

Reply:
We didn't even have anyone asking for this really, just wanted to expose it for Windows. Why would they be better as shim options? Or are you saying in addition to this and this would override the shim option like how we normally do? If certain workloads preferred a certain dump type it'd be hard to configure if it was node wide and required a reboot to change.

Reply:
Oh realized you selected both of these and I only touched on the dump type. For the dump location that makes more sense.

Reply:
Yeah, the shim options would be over written if these annotations are set.

Reply:
I'm not fond of the dump type but the process dump location sounds sweet, I'll add

Reply:
I don't see why the process dump location couldn't also be a shim option? You would just override with the annotation? 

I guess it depends: are there workloads where they will want either of these options set as a default all the time?

Reply:
I never said it couldn't, I'm agreeing with you 🙂. I'm saying I don't think we need one for dump type, I'm not sure we need the annotation even, just thought it'd be nice to expose. I could see some folks wanting mini dumps for certain workloads or full for others though.

Reply:
Sorry, I meant the dump type :) 

Is this something just for debugging? If that's the case, shim options don't really make sense to use. Otherwise, we should see how often they might want to override a "default" value to see if shim options make sense.

Reply:
No, not for debugging, just an extra configurable option on Windows that seemed like it'd make sense to be a choice. Would it make more sense to add these later if someone requests? Not keen on adding this now and then we'd be stuck with it in the name of a breaking change if we ever wanted to remove the options.